### PR TITLE
:gear:  Add logic

### PR DESCRIPTION
**Pull request by SimonGideon**
- [ ] Handle events in a React app.
- [ ] Use React life cycle methods.
- [ ] Understand the mechanism of lifting the state up.

Thanks for the review :slightly_smiling_face:

### CODE REVIEW

**Review by ha-manel — COMMENTED** [review]
# STATUS: INVALID 🔴 
Hi 
I'm sure you put a lot of effort into this project but this PR is invalid because you're not following `Gitflow` correctly 😓 the PR should be from the feature branch to the `dev` branch not to the `main` branch 👍 

Kindly fix the issue and submit the PR for review again 👍 
Best of luck!

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SimonGideon** [comment]
@ha-manel Thanks for that,,  :slightly_smiling_face: The code review Heading is terrifying

giphy (1)

**Comment by ha-manel** [comment]
@SimonGideon haha thanks for the feedback, I'll make sure to change it into something softer 🤣

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,78 +1,89 @@
...
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+    err: null,
+  });
+  const handleClick = (e) => {
+    setState((prevState) => calculate(prevState, e.target.textContent));
+  };
```
Comment (emmiiorji) [inline src/components/Calculator.js:16]: Just like you rightly had earlier, it is [required](https://github.com/microverseinc/curriculum-react-redux/blob/e404f92ae7b446ece2b5081c85779f0d9ab59731/math-magicians/project_components.md?plain=1#L40) to use class components for this project. Kindly continue using class components.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,78 +1,89 @@
-import React from 'react';
+/* eslint-disable react/destructuring-assignment */
```
Comment (emmiiorji) [inline src/components/Calculator.js:1]: As much as possible, kindly avoid disabling linter checks in line. They represent a coding standard you share with your team and most of the time help to enforce best practices.

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @SimonGideon, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

A little bit more

## Good Highlights✔️

- Linter checks are passing
- Good commit messages
- Adhered to Gitflow


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (SimonGideon) [reply src/components/Calculator.js:16]: @emmiiorji This is awesome to observe the little changes to make my project meet the requirements. The suggested changes are made on the preceding commits. :slightly_smiling_face:

File: src/components/Calculator.js (hunk 2 above)
Comment (SimonGideon) [reply src/components/Calculator.js:1]: @emmiiorji Noted this. I'll put it into practise on future commits.

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @SimonGideon , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- GitHub flow is used correctly. ✔
- All linters checks are ok. ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SimonGideon** [comment]
Thanks for the review @ridwanediallo 
giphy

**Pull request merged**